### ISSUE TITLE
Loading a model in SharpGLTF.Toolkit seems to create duplicate meshes

### ISSUE THREAD

**Issue by cyril-perrin** [vpenades/SharpGLTF#123]
Hello, I’m using SharpGLTF.Core to :
1. Open an existing model
`var model = ModelRoot.Load("model.glb");`
2. Create a new scene
`var outputScene = model.UseScene("Working Scene");`
3. Copy several times a selection of nodes from the default scene to the new working scene
```
for (...)
{
   ...
   var newNode = outputScene.CreateNode("Copy of " + node.Name);
   newNode.Skin = node.Skin;
   newNode.Mesh = node.Mesh;
   newNode.LocalTransform = node.LocalTransform * transformationMatrix;
}
```
4. Set my working scene as the default scene
`model.DefaultScene = outputScene;`

When I save the model, the resulting GLB is hardly larger than the source one, since all the nodes I created point to the same mesh (confirmed in  Gestaltor). This is perfect! 

However… when I load my working scene in the Toolkit using SceneBuilder.CreateFrom(), then save the GLB, the file size is much larger since all the nodes now point to their own duplicate of the same mesh.

Is this the expected behavior? Is there a way to prevent the creation of these extra meshes?

Thanks.

**Comment by vpenades** [comment]
> Is this the expected behavior? 

No, SceneBuilder has a mechanism to identify identical meshes and reuse them.

But this mechanism operate when you convert to schema gift. Are you converting multiple scenes within a single model?

**Comment by cyril-perrin** [comment]
Yes I'm using the code you propose here: https://github.com/vpenades/SharpGLTF/issues/79#issuecomment-734314259
```
var merged = new SceneBuilder();
var suzanne = SceneBuilder.Load("Suzanne.glb");
merged.AddScene(suzanne, System.Numerics.Matrix4x4.Identity);
merged.ToGltf2().SaveGLB("Suzanne2.glb");
```
with the attached model ([Suzanne.zip](https://github.com/vpenades/SharpGLTF/files/8140838/Suzanne.zip)).
Suzanne.glb is around 2.5MB, Suzanne2.glb is 114MB.
Thanks.

**Comment by vpenades** [comment]
yeah, that's the source of the problem.

When I developed these classes, I was not expecting many people to use multiple scenes, so all the interal optimizations to identify shared resources only work for a single scene.

I'll find a solution for this, and I'll extend the API so sceneBuilder can load multiple scenes simultaneously while sharing the resources.

**Comment by vpenades** [comment]
I've commited some changes that will fix the problem, you'll have to clone the repo and use the projects for a while until next release.

use case example can be seen [here.](https://github.com/vpenades/SharpGLTF/blob/034a40b1a1c43dc2810704f206c3693f1159078d/tests/SharpGLTF.Toolkit.Tests/Scenes/SceneBuilderTests.cs#L904)

**Comment by cyril-perrin** [comment]
Thank you for your quick reply.
I adapted my code for your latest commit. It is now:
```
var merged = new SceneBuilder();
var suzanne = SceneBuilder.LoadDefaultScene("Suzanne.glb");
merged.AddScene(suzanne, System.Numerics.Matrix4x4.Identity);
merged.ToGltf2().SaveGLB("Suzanne2.glb");
```
Using the same model ([Suzanne.zip](https://github.com/vpenades/SharpGLTF/files/8140838/Suzanne.zip)), I unfortunately experience almost the same result as before (Suzanne2.glb weights 112MB now) and in addition, Gestaltor logs "Error | Could not bind invalid texture".

Sorry if I'm missing something!

**Comment by cyril-perrin** [comment]
Hello, I tested 'master' on 059a947 and I confirm that AddScene() doesn’t duplicate the meshes anymore, thanks!
However, I believe something has been broken. 
```
var suzanne = ModelRoot.Load("glTF/Suzanne.gltf", ValidationMode.TryFix);
suzanne.SaveGLB("Suzanne.glb");
```
Just by running this piece of code with the original Suzanne model ([here](https://github.com/KhronosGroup/glTF-Sample-Models/tree/master/2.0/Suzanne)), I notice that:
- The generated glb file only weights  577KB whereas the files in the glTF folder weight 2.5MB
- Gestaltor complains that it could not bind an invalid texture
